src/Data/Text/Internal/Encoding/Utf8.hs

```diff
@@ -34,19 +34,18 @@ module Data.Text.Internal.Encoding.Utf8
     , validate3
     , validate4
     -- * Naive decoding
-    , DecoderResult(..)
-    , DecoderState(..)
-    , CodePoint(..)
-    , utf8DecodeStart
-    , utf8DecodeContinue
+    , Utf8CodePointState
+    , utf8StartState
+    , updateUtf8State
+    , isUtf8StateIsComplete
```

Comment:
While the module is declared as internal, let's nevertheless describe the change in the changelog.

Reply:
It turns out those were mostly cosmetic changes so I reverted them.